Hyperborder no longer works with hyper-2.0.0 canary 15

```
bundle.js:1 Error: Cannot find module '/Users/XXX/.hyper_plugins/node_modules/hyperborder'
    at Module._resolveFilename (module.js:470:15)
    at Function.Module._resolveFilename (/Applications/Hyper.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:418:25)
    at Function.c._load (file:///Applications/Hyper.app/Contents/Resources/app.asar/renderer/bundle.js:1:8001)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at D.e.plugins.concat.map (file:///Applications/Hyper.app/Contents/Resources/app.asar/renderer/bundle.js:1:8948)
    at Array.map (native)
    at D (file:///Applications/Hyper.app/Contents/Resources/app.asar/renderer/bundle.js:1:8899)
    at Object.<anonymous> (file:///Applications/Hyper.app/Contents/Resources/app.asar/renderer/bundle.js:1:10452)
```

--- a/index.js
+++ b/index.js
@@ -7,16 +7,17 @@ let unloadAnimator = null;
 
 module.exports.onRendererWindow = async window => {
   const browserWindow = remote.getCurrentWindow();
+  const htmlElement = window.document.documentElement;
 
-  browserWindow.on('blur', () => window.document.documentElement.classList.add('blurred'));
-  browserWindow.on('focus', () => window.document.documentElement.classList.remove('blurred'));
+  browserWindow.on('blur', () => htmlElement.classList.add('blurred'));
+  browserWindow.on('focus', () => htmlElement.classList.remove('blurred'));
 
   if (!browserWindow.isFocused()) {
-    window.document.documentElement.classList.add('blurred');
+    htmlElement.classList.add('blurred');
   }
 
   if (await isElevated()) {
-    window.document.documentElement.classList.add('elevated');
+    htmlElement.classList.add('elevated');
   }
 
   const config = window.config.getConfig();
@@ -32,74 +33,99 @@ module.exports.onUnload = async () => {
   }
 };
 
-module.exports.decorateConfig = config => {
+module.exports.reduceUI = (state, {type, config}) => {
   const defaultColors = ['#fc1da7', '#fba506'];
+  switch (type) {
+    case 'CONFIG_LOAD':
+    case 'CONFIG_RELOAD':
+      return state.set('hyperBorder', Object.assign({
+        animate: false,
+        backgroundColor: config.backgroundColor,
+        borderWidth: '4px',
+        borderColors: defaultColors,
+        adminBorderColors: (config.hyperBorder && config.hyperBorder.borderColors) || defaultColors,
+        blurredAdminColors: (config.hyperBorder && (config.hyperBorder.blurredColors || config.hyperBorder.adminBorderColors)) || defaultColors,
+        blurredColors: defaultColors,
+        borderAngle: '180deg'
+      }, config.hyperBorder));
+    default:
+      return state;
+  }
+};
 
-  const configObj = Object.assign({
-    animate: false,
-    borderWidth: '4px',
-    borderColors: defaultColors,
-    adminBorderColors: (config.hyperBorder && config.hyperBorder.borderColors) || defaultColors,
-    blurredAdminColors: (config.hyperBorder && (config.hyperBorder.blurredColors || config.hyperBorder.adminBorderColors)) || defaultColors,
-    blurredColors: defaultColors,
-    borderAngle: '180deg'
-  }, config.hyperBorder);
+module.exports.mapHyperState = ({ui: {hyperBorder}}, map) => Object.assign({}, map, {
+  hyperBorder: Object.assign({}, hyperBorder)
+});
 
-  return Object.assign({}, config, {
-    css: `
-      ${config.css || ''}
-      {
-        --border-width: ${configObj.borderWidth};
-        --border-angle: ${configObj.animate ? '269deg' : configObj.borderAngle};
-        --background-color: ${config.backgroundColor || '#000'};
-        --border-color: ${config.borderColor};
-        --border-colors: ${getBorderColors(configObj.borderColors).join(',')};
-        --admin-colors: ${getBorderColors(configObj.adminBorderColors).join(',')};
-        --blurred-colors: ${getBorderColors(configObj.blurredColors).join(',')};
-        --blurred-admin-colors: ${getBorderColors(configObj.blurredAdminColors).join(',')};
-        background: linear-gradient(var(--border-angle), var(--border-colors));
-        top: 0;
-        bottom: 0;
-        left: 0;
-        right: 0;
-        position: fixed;
-        border-radius: var(--border-width);
-      }
-      .hyper_main {
-        background-color: var(--background-color);
-        top: var(--border-width);
-        bottom: var(--border-width);
-        left: var(--border-width);
-        right: var(--border-width);
-        border-radius: var(--border-width);
-      }
-      .hyper_main .header_header {
-        top: var(--border-width);
-        left: var(--border-width);
-        right: var(--border-width);
-      }
-      .hyper_main .header_windowHeader {
-        top: var(--border-width);
-        left: var(--border-width);
-        right: var(--border-width);
-        width: calc(100% - var(--border-width) - var(--border-width)); 
-      }
-      .hyper_main .header_hamburgerMenuLeft {
-        top: var(--border-width);
-        left: var(--border-width);        
-      }
-      // blurred is not working for now
-      /*
-      .elevated {
-        background: linear-gradient(var(--border-angle), var(--admin-colors));
-      }
-      .blurred {
-        background: linear-gradient(var(--border-angle), var(--blurred-colors));
-      }
-      .blurred.elevated {
-        background: linear-gradient(var(--border-angle), var(--blurred-admin-colors));
-      }
-      */
-    `
-  });
+module.exports.decorateHyper = (Hyper, {React}) => {
+  return class extends React.Component {
+    constructor(props) {
+      super(props);
+      this.initializeAnimator();
+    }
+
+    initializeAnimator() {
+      // TODO init Animator here
+    }
+
+    render() {
+      return React.createElement('div', {
+        id: 'hyperborder'
+      }, [
+        React.createElement(Hyper, this.props),
+        React.createElement('style', {}, `
+        #hyperborder {
+          --border-width: ${this.props.hyperBorder.borderWidth};
+          ${this.props.hyperBorder.animate ? '' : '--border-angle: ' + this.props.hyperBorder.borderAngle + ';'}
+          --background-color: ${this.props.hyperBorder.backgroundColor || '#000'};
+          --border-colors: ${getBorderColors(this.props.hyperBorder.borderColors).join(',')};
+          --admin-colors: ${getBorderColors(this.props.hyperBorder.adminBorderColors).join(',')};
+          --blurred-colors: ${getBorderColors(this.props.hyperBorder.blurredColors).join(',')};
+          --blurred-admin-colors: ${getBorderColors(this.props.hyperBorder.blurredAdminColors).join(',')};
+        }
+        #hyperborder #hyper {
+          background: linear-gradient(var(--border-angle), var(--border-colors));
+          top: 0;
+          bottom: 0;
+          left: 0;
+          right: 0;
+          position: fixed;
+          border-radius: var(--border-width);
+        }
+        #hyperborder #hyper .hyper_main {
+          background-color: var(--background-color);
+          top: var(--border-width);
+          bottom: var(--border-width);
+          left: var(--border-width);
+          right: var(--border-width);
+          border-radius: var(--border-width);
+        }
+        #hyperborder #hyper .hyper_main .header_header {
+          top: var(--border-width);
+          left: var(--border-width);
+          right: var(--border-width);
+        }
+        #hyperborder #hyper .hyper_main .header_windowHeader {
+          top: var(--border-width);
+          left: var(--border-width);
+          right: var(--border-width);
+          width: calc(100% - var(--border-width) - var(--border-width));
+        }
+        #hyperborder #hyper .hyper_main .header_hamburgerMenuLeft {
+          top: var(--border-width);
+          left: var(--border-width);
+        }
+        html.elevated #hyperborder #hyper {
+          background: linear-gradient(var(--border-angle), var(--admin-colors));
+        }
+        html.blurred #hyperborder #hyper {
+          background: linear-gradient(var(--border-angle), var(--blurred-colors));
+        }
+        html.blurred.elevated #hyperborder #hyper {
+          background: linear-gradient(var(--border-angle), var(--blurred-admin-colors));
+        }
+        `)
+      ]);
+    }
+  };
 };

--- a/index.js
+++ b/index.js
@@ -47,7 +47,8 @@ module.exports.decorateConfig = config => {
 
   return Object.assign({}, config, {
     css: `
-      html {
+      ${config.css || ''}
+      {
         --border-width: ${configObj.borderWidth};
         --border-angle: ${configObj.animate ? '269deg' : configObj.borderAngle};
         --background-color: ${config.backgroundColor || '#000'};
@@ -56,32 +57,13 @@ module.exports.decorateConfig = config => {
         --admin-colors: ${getBorderColors(configObj.adminBorderColors).join(',')};
         --blurred-colors: ${getBorderColors(configObj.blurredColors).join(',')};
         --blurred-admin-colors: ${getBorderColors(configObj.blurredAdminColors).join(',')};
-      }
-      html {
-        height: 100%;
         background: linear-gradient(var(--border-angle), var(--border-colors));
+        top: 0;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        position: fixed;
         border-radius: var(--border-width);
-        overflow: hidden;
-      }
-      html.elevated {
-        background: linear-gradient(var(--border-angle), var(--admin-colors));
-      }
-      html.blurred {
-        background: linear-gradient(var(--border-angle), var(--blurred-colors));
-      }
-      html.blurred.elevated {
-        background: linear-gradient(var(--border-angle), var(--blurred-admin-colors));
-      }
-      body {
-        position: absolute;
-        top: var(--border-width);
-        bottom: var(--border-width);
-        left: var(--border-width);
-        right: var(--border-width);
-        border-radius: var(--border-width);
-      }
-      ${config.css || ''}
-      #mount {
       }
       .hyper_main {
         background-color: var(--background-color);
@@ -89,34 +71,35 @@ module.exports.decorateConfig = config => {
         bottom: var(--border-width);
         left: var(--border-width);
         right: var(--border-width);
-        border-width: 0px;
+        border-radius: var(--border-width);
       }
       .hyper_main .header_header {
         top: var(--border-width);
         left: var(--border-width);
         right: var(--border-width);
       }
-      .hyper_main .tabs_list {
-        border-bottom-color: var(--border-color);
-        border-top-left-radius: var(--border-width);
-        border-top-right-radius: var(--border-width);
-      }
-      .hyper_main .tab_tab:last-child {
-        border-top-right-radius: var(--border-width);
-      }
-      .hyper_main .terms_terms {
-        border-radius: 0 0 var(--border-width) var(--border-width);
-        bottom: var(--border-width);
+      .hyper_main .header_windowHeader {
+        top: var(--border-width);
         left: var(--border-width);
         right: var(--border-width);
+        width: calc(100% - var(--border-width) - var(--border-width)); 
       }
-      .hyper_main .terms_term {
-        margin-top: var(--border-width);
+      .hyper_main .header_hamburgerMenuLeft {
+        top: var(--border-width);
+        left: var(--border-width);        
       }
-      .header_windowHeaderWithBorder {
-        left: var(--border-width);
-        width: calc(100% - var(--border-width) - var(--border-width));
+      // blurred is not working for now
+      /*
+      .elevated {
+        background: linear-gradient(var(--border-angle), var(--admin-colors));
       }
+      .blurred {
+        background: linear-gradient(var(--border-angle), var(--blurred-colors));
+      }
+      .blurred.elevated {
+        background: linear-gradient(var(--border-angle), var(--blurred-admin-colors));
+      }
+      */
     `
   });
 };

--- a/index.js
+++ b/index.js
@@ -82,8 +82,6 @@ module.exports.decorateHyper = (Hyper, {React}) => {
           --admin-colors: ${getBorderColors(this.props.hyperBorder.adminBorderColors).join(',')};
           --blurred-colors: ${getBorderColors(this.props.hyperBorder.blurredColors).join(',')};
           --blurred-admin-colors: ${getBorderColors(this.props.hyperBorder.blurredAdminColors).join(',')};
-        }
-        #hyperborder #hyper {
           background: linear-gradient(var(--border-angle), var(--border-colors));
           top: 0;
           bottom: 0;
@@ -92,7 +90,7 @@ module.exports.decorateHyper = (Hyper, {React}) => {
           position: fixed;
           border-radius: var(--border-width);
         }
-        #hyperborder #hyper .hyper_main {
+        #hyperborder .hyper_main {
           background-color: var(--background-color);
           top: var(--border-width);
           bottom: var(--border-width);
@@ -100,28 +98,28 @@ module.exports.decorateHyper = (Hyper, {React}) => {
           right: var(--border-width);
           border-radius: var(--border-width);
         }
-        #hyperborder #hyper .hyper_main .header_header {
+        #hyperborder .hyper_main .header_header {
           top: var(--border-width);
           left: var(--border-width);
           right: var(--border-width);
         }
-        #hyperborder #hyper .hyper_main .header_windowHeader {
+        #hyperborder .hyper_main .header_windowHeader {
           top: var(--border-width);
           left: var(--border-width);
           right: var(--border-width);
           width: calc(100% - var(--border-width) - var(--border-width));
         }
-        #hyperborder #hyper .hyper_main .header_hamburgerMenuLeft {
+        #hyperborder .hyper_main .header_hamburgerMenuLeft {
           top: var(--border-width);
           left: var(--border-width);
         }
-        html.elevated #hyperborder #hyper {
+        html.elevated #hyperborder {
           background: linear-gradient(var(--border-angle), var(--admin-colors));
         }
-        html.blurred #hyperborder #hyper {
+        html.blurred #hyperborder {
           background: linear-gradient(var(--border-angle), var(--blurred-colors));
         }
-        html.blurred.elevated #hyperborder #hyper {
+        html.blurred.elevated #hyperborder {
           background: linear-gradient(var(--border-angle), var(--blurred-admin-colors));
         }
         `)

--- a/lib/animator.js
+++ b/lib/animator.js
@@ -11,7 +11,7 @@ let intervalId = null;
  * gradient border
  * @param {object} window - the window object
  * @param {object} browserWindow - the browserWindow object
- * @return {object} contains function to unload animator
+ * @return {function} function to unload animator
  */
 module.exports.createAnimator = (window, browserWindow) => {
   const updateBorderAngle = () => {

--- a/test/index.test.js
+++ b/test/index.test.js
@@ -3,19 +3,6 @@ const {oneLine} = require('common-tags');
 const {createMockElectron, createMockWindow, createMockBrowserWindow} = require('./helpers');
 const proxyquire = require('proxyquire').noCallThru();
 
-// --- decorateConfig() ---
-
-test(oneLine`
-  decorateConfig() doesn't throw when there isn't
-  a hyperBorder config section`,
-  t => {
-    const {decorateConfig} = proxyquire('../index', {
-      electron: {}
-    });
-    t.notThrows(() => decorateConfig({}));
-  }
-);
-
 // --- onRendererWindow() ---
 
 test(oneLine`

